vault/dynamic_system_view.go

```diff
@@ -181,6 +181,9 @@ func (d dynamicSystemView) LocalMount() bool {
 // in read mode.
 func (d dynamicSystemView) ReplicationState() consts.ReplicationState {
 	state := d.core.ReplicationState()
+	if d.core.standby.Load() {
```

Comment:
Which is to say, should https://github.com/openbao/openbao/pull/1986/files#r2450003133 handle this rather than setting it here.